chat/src/agent/tools.py

```diff
@@ -63,3 +64,25 @@ def aggregate(agg_field: str, term_field: str, term: str):
         return json.dumps(response, default=str), response
     except Exception as e:
         return json.dumps({"error": str(e)}), None
+
+@tool(response_format="content_and_artifact")
+def retrieve_documents(doc_ids: List[str]):
+    """
+    Retrieve documents from the OpenSearch index based on a list of document IDs. 
+
+    Use this instead of the search tool if the user has provided docs for context
+    and you need the full metadata in order to answer their question.
+
```

Comment:
I added this line

```
Provide an answer to their question based on the metadata of the documents.
```

Not sure if it really helps or just confirmation bias on my parth